test/v5/kyberTradeLogic.js

```diff
@@ -690,63 +698,513 @@ contract('KyberTradeLogic', function(accounts) {
+                    }   
+                }
             });
 
             it("T2T, mask in hint | split hint", async() => {
+                for (takerFeeBps of takerFeeArray) {
+                    for (platformFeeBps of platformFeeArray) {
+                        fees = [takerFeeBps, platformFeeBps];
+                        //search with no fees
+                        hintedReserves = await getHintedReserves(
+                            tradeLogic, reserveInstances,
+                            MASK_IN_HINTTYPE, undefined, undefined, srcQty,
+                            SPLIT_HINTTYPE, undefined, undefined, ethSrcQty,
+                            srcToken.address, destToken.address
+                            );
 
+                        bestSellReserve = await nwHelper.getBestReserveAndRate(hintedReserves.reservesT2E.reservesForFetchRate, srcToken.address, ethAddress, srcQty, takerFeeBps); 
+                        reserveRatesE2T = hintedReserves.reservesE2T.reservesForFetchRate.map(reserve => reserve.rate);
+
+                        expectedTradeResult = getTradeResult(
+                            srcDecimals, [bestSellReserve], [bestSellReserve.rateNoFee], [],
+                            destDecimals, hintedReserves.reservesE2T.reservesForFetchRate, reserveRatesE2T, hintedReserves.reservesE2T.splits,
+                            srcQty, takerFeeBps, platformFeeBps);
+
+                        //T2E
+                        expectedReserves = expectedReserves.concat(bestSellReserve.address);
+                        expectedRates = expectedRates.concat(bestSellReserve.rateNoFee);
+                        expectedSplitValuesBps = expectedSplitValuesBps.concat(BPS);
+                        expectedFeePaying = expectedFeePaying.concat(bestSellReserve.isFeePaying);
+
+                        //E2T
+                        expectedReserves = expectedReserves.concat(hintedReserves.reservesE2T.reservesForFetchRate.map(reserve => reserve.address));
+                        expectedRates = expectedRates.concat(reserveRatesE2T);
+                        expectedSplitValuesBps = expectedSplitValuesBps.concat(hintedReserves.reservesE2T.splits);
+                        expectedFeePaying = expectedFeePaying.concat(hintedReserves.reservesE2T.reservesForFetchRate.map(reserve => reserve.isFeePaying));
+                        
+                        actualResult = await tradeLogic.calcRatesAndAmounts(srcToken.address, destToken.address, srcQty, fees, hintedReserves.hint);
+                        compareResults(expectedTradeResult, expectedReserves, expectedRates, expectedSplitValuesBps, expectedFeePaying, actualResult);
+                    }   
+                }
             });
 
             it("T2T, mask out hint | no hint", async() => {
+                for (takerFeeBps of takerFeeArray) {
```

Comment:
wow all combinations?
see if can reduce number of lines using helper function. so make the code shorter.
or looping combinations